### ISSUE TITLE
Having no joy with this component - "exports is not defined"

### ISSUE THREAD

**Issue by squeezeboxhuf** [gtm-nayan/svelte-pdfjs#8]
I have a new sveltekit project, I have added svelte-pdfjs (using: npm -i svelte-pdfjs), but my page with the component which is taken more or less from your  routes/index.svelte just gives me an error 500:-

```
exports is not defined
ReferenceError: exports is not defined
    at http://localhost:3000/node_modules/svelte-pdfjs/node_modules/pdfjs-dist/lib/pdf.js?v=96c44fcc:24:23
```

I can see pdfjs-dist in my node_modules @ 2.12.313. I've tried earlier version but still get the same. 

My page.svelte contains:
```svelte
<script lang="ts">
	import { Document, Page, PDFJS, preferThisHeight } from 'svelte-pdfjs';
	import type { MultipleOf90 } from 'svelte-pdfjs';
	import workerSrc from 'pdfjs-dist/build/pdf.worker.js?url';
	import { browser } from '$app/env';
	import { selectedPart } from '$lib/stores';

	import { MEDIA_URL } from '$lib/variables';

	if (PDFJS.GlobalWorkerOptions) {
		PDFJS.GlobalWorkerOptions.workerSrc = workerSrc;
	}

	let scale: number = 1;
	let num: number = 1;
	let maxPages = 1;
	let renderTextLayer: boolean;
	let targetHeight = 500;
	let rotation: MultipleOf90 = 0;

	const file = `${MEDIA_URL}${$selectedPart.partfile}`;
	console.log(file);
</script>

{#if browser}
	<Document
		{file}
		on:loadsuccess={(e) => console.log((maxPages = e.detail.numPages))}
		on:loaderror={console.log}
	>
		<div>
			<Page
				{scale}
				{num}
				{renderTextLayer}
				{rotation}
				getViewport={preferThisHeight(targetHeight)}
			/>
		</div>
	</Document>
{/if}
```

What am I missing? Thanks.
Hamish

**Comment by gtm-nayan** [comment]
@squeezeboxhuf can you check which version of svelte-pdfjs you're using? I was experimenting with some stuff on the alpha versions so if you have those it'll not work. The latest stable version 0.3.4 should work.

**Comment by squeezeboxhuf** [comment]
I have made some progress! I discovered that I had to have a sveltekit project **with** TypeScript, then I modded my svelte.config.js and tsconfig.json to match your additional aliases, then copied the component parts from this repository src/lib into my src/lib folder and Hey Presto, I see a PDF document!

**Comment by gtm-nayan** [comment]
I'm glad you found a way to make it work somehow but that's not how the package is meant to be used. 

If it's not too much of a bother to you, can you install version 0.3.4 with `npm i svelte-pdfjs@0.3.4 pdfjs-dist` and see if the issue persists? Thanks. 🙂

**Comment by squeezeboxhuf** [comment]
Yes that did the trick! So I reverted my tsconfig and svelte.config, removed the src/lib additions and I am still getting PDFs displayed! Thank you. 

One extra question! I have some multi-page documents, so I am using a {#each} loop for maxPages numbers to repeat the `<Page>` component within the` <Document>` ... like this...


`{#if browser}`
`		<Document`
`			{file}`
`			on:loadsuccess={(e) => console.log((maxPages = e.detail.numPages))}`
`			on:loaderror={console.log}`
`		>`
`			{#each Array.from({ length: maxPages }, (_, i) => i + 1) as num}`
`				<p class="separator">page {num} || <span on:click={handleRotate}>rotate</span></p>`
`				<div>`
`					<Page`
`						{scale}`
`						{num}`
`						{renderTextLayer}`
`						{rotation}`
`						getViewport={preferThisWidth(targetWidth)}`
`					/>`
`				</div>`
`			{/each}`
`		</Document>`
`	{/if}`


But this results in an error after the first page (first page displays OK) : "PageCanvas.svelte? [sm]:9 Uncaught (in promise) TypeError: Cannot set properties of undefined (setting 'width')  at render_page (PageCanvas.svelte? [sm]:9:17)". So I put the each loop around the `<Document>` component and it's OK. Effectively I'm creating a `<Document>` with a` <Page>` within it, for each page. Is that what you'd expect to be needed? My first presumption was to iterate just the `<Page>` elements.

I've discovered it might be my page separator paragraph interfering... It's OK to iterate just the `<Page>` as long as I don't put any other objects in the loop... I've tried putting separator para before and after but it still seems to upset the subsequent pages. 

For now, I've left it iterating each containing `<Document>` and` <Page>`.

Thanks

**Comment by gtm-nayan** [comment]
I just published 0.3.7 to npm which should fix it.

@squeezeboxhuf Upgrade and let me know if it works or not.

**Comment by squeezeboxhuf** [comment]
Yes - that's great. Thank you.
